File: exercises/imperative-to-functional/04/example-solution.py
import collections
import json
import sys


data = json.loads(sys.stdin.read())

frequencies = collections.Counter(exam['location'] for exam in data['exams'])
result = sorted(location for location, frequency in frequencies.items() if frequency > data['room_capacities'][location])

print("\n".join(str(x) for x in result))

File: exercises/imperative-to-functional/04/example-solution.rb
require 'json'

rid, = ARGV

data = JSON.parse(STDIN.read)

locFreq = data['exams'].map { |exam| exam['location'] }.tally
locs = locFreq.keys
caps = data['room_capacities']
locs.select { |loc| locFreq[loc] > caps[loc] }
    .sort
    .then { |r| puts r }